Jint/Engine.cs

```diff
@@ -761,7 +765,12 @@ public JsValue GetValue(JsValue scope, string propertyName)
         }
 
         //  http://www.ecma-international.org/ecma-262/5.1/#sec-10.5
-        public void DeclarationBindingInstantiation(DeclarationBindingType declarationBindingType, IEnumerable<FunctionDeclaration> functionDeclarations, IEnumerable<VariableDeclaration> variableDeclarations, FunctionInstance functionInstance, JsValue[] arguments)
+        public void DeclarationBindingInstantiation(
+            DeclarationBindingType declarationBindingType,
+            IList<FunctionDeclaration> functionDeclarations,
```

Comment:
IList gives us indexing but not allocation free foreacher, so using indexing

Reply:
Is there a reason to _not_ use `List<T>` here?

Reply:
In particular, I'm thinking about bound check elimintations.

Reply:
The problem here is (like other places too) that we are dealing with public API. Changing to concrete implementations might not be the prettiest thing, even though best performing. I think @sebastienros  can be the judge of this.